### PR TITLE
feat: Add support for allowing EFA network interfaces

### LINKED ISSUE #1880: self_managed_node_group creates invalid launch template when using network_interfaces
## Description

When using the EKS module to create a self managed node group if you define a network_interface in the self_managed_node_group sub module it will create an invalid template per issue https://github.com/hashicorp/terraform-provider-aws/issues/4570. It looks like there needs to be a conditional for the vpc_security_group_ids value setting it to null when a network_interface is provided. I was able to get past the error and add the EFA interface_type I wanted by adding the following.
```
git diff
diff --git a/modules/self-managed-node-group/main.tf b/modules/self-managed-node-group/main.tf
index 1b7ca66..52bcb81 100644
--- a/modules/self-managed-node-group/main.tf
+++ b/modules/self-managed-node-group/main.tf
@@ -57,7 +57,7 @@ resource "aws_launch_template" "this" {
   key_name      = var.key_name
   user_data     = module.user_data.user_data

-  vpc_security_group_ids = compact(concat([try(aws_security_group.this[0].id, "")], var.vpc_security_group_ids))
+  vpc_security_group_ids = length(var.network_interfaces) > 0 ? null :  compact(concat([try(aws_security_group.this[0].id, "")], var.vpc_security_group_ids))

   default_version                      = var.launch_template_default_version
   update_default_version               = var.update_launch_template_default_version
@@ -197,6 +197,7 @@ resource "aws_launch_template" "this" {
     for_each = var.network_interfaces
     content {
       associate_carrier_ip_address = lookup(network_interfaces.value, "associate_carrier_ip_address", null)
+      interface_type               = lookup(network_interfaces.value, "interface_type", null)
       associate_public_ip_address  = lookup(network_interfaces.value, "associate_public_ip_address", null)
       delete_on_termination        = lookup(network_interfaces.value, "delete_on_termination", null)
       description                  = lookup(network_interfaces.value, "description", null)
```
Although I ran into a new error. I was able to get past it by setting the variable to not create the Cloudwatch log group. I am not sure how the change is related to this new error.

`│ Error: Creating CloudWatch Log Group failed: ResourceAlreadyExistsException: The specified log group already exists:  The CloudWatch Log Group '/aws/eks/tf-west-test-vpc/cluster' already exists.`

## Versions
 - Terraform:
1.1.5
 - Provider(s):
AWS 3.74.2
 - Module:
 EKS 18.2.7

## Reproduction
Create an EKS cluster using the EKS module with a self_managed_node_group that has a network_interface defined. This will create a clean plan but when you apply the the autoscaling group will fail to create due to the issue listed above (When a network interface is defined for a launch template the security groups must be defined in the network interfaces section and not with vpc_security_group_ids in the main block of the launch template.)

### Code Snippet to Reproduce

```

self_managed_node_groups = { system = {
            name = "system"
            instance_type = var.node_type
            ami_id = var.ami.id
            desired_size = 2
            min_size = 1
            max_size = 4
            key_name = ""
            bootstrap_extra_args = "--kubelet-extra-args '--node-labels=node-role.jarvice.io/jarvice-system=true,node-pool.jarvice.io/jarvice-system=jxesystem --register-with-taints=node-role.jarvice.io/jarvice-system=true:NoSchedule'"
            network_interfaces = [
                    {
                      device_index = 0
                      associate_public_ip_address = true
                      interface_type = "efa"
                      security_groups = ["${aws_security_group.ssh.id}"] 
                    }
                ]
            enable_bootstrap_user_data = true
            pre_bootstrap_user_data = <<EOF
# pre_userdata (executed before kubelet bootstrap and cluster join)
# Add authorized ssh key
echo "${var.ssh_public_key}" >>/home/ec2-user/.ssh/authorized_keys
EOF
        }
}
```

## Expected behavior
Successful creation of a self managed node group with an EFA interface

## Actual behavior
Error on terraform apply.
│ Error: Error creating Auto Scaling Group: InvalidQueryParameter: Invalid launch template: When a network interface is provided, the security groups must be a part of it.

### PR DESCRIPTION
**Pull request by bryantbiggs**
## Description
- Add support for allowing EFA network interfaces; no 
- Additional interface attributes were not added due to requiring v4 of provider - will come later at next breaking change

## Motivation and Context
- Closes #1880
- Closes #1907

## Breaking Changes
- No

## How Has This Been Tested?
- [x] I have updated at least one of the `examples/*` to demonstrate and validate my change(s)
- [x] I have tested and validated these changes using one or more of the provided `examples/*` projects

- [x] I have executed `pre-commit run -a` on my pull request

### CODE REVIEW

**Comment by bryantbiggs** [comment]
If you get time 🙏🏼 @antonbabenko

File: examples/self_managed_node_group/main.tf [hunk 1]
```diff
@@ -146,6 +146,37 @@ module "eks" {
       }
     }
 
+    efa = {
+      min_size     = 1
+      max_size     = 2
+      desired_size = 1
+
+      # aws ec2 describe-instance-types --region eu-west-1 --filters Name=network-info.efa-supported,Values=true --query "InstanceTypes[*].[InstanceType]" --output text | sort
+      instance_type = "c5n.9xlarge"
```
Comment (antonbabenko) [inline examples/self_managed_node_group/main.tf:155]: Looks like the cheapest instance type among the most expensive ones! That's what we need in examples :)

**Comment by antonbabenko** [comment]
This PR is included in version 18.18.0 :tada:

File: examples/self_managed_node_group/main.tf (hunk 1 above)
Comment (bryantbiggs) [reply examples/self_managed_node_group/main.tf:155]: ya I know 😅 , I might end up commenting this out and putting a note to check periodically

**Pull request merged**